src/conn/conn_dhandle.c

```diff
@@ -313,6 +313,17 @@ __wt_conn_dhandle_close(WT_SESSION_IMPL *session, bool final, bool mark_dead)
     btree = is_btree ? dhandle->handle : NULL;
 
     if (is_btree) {
+        /*
+         * !! Check if we have exclusive access here, else a transaction can start on this btree
+         * after this check !!
+         *
+         * Don't close the btree if there are changes not yet globally visible.
+         */
+        WT_ASSERT_ALWAYS(session, btree->max_upd_txn != WT_TXN_ABORTED,
+          "Assert failure: session: %s: btree->max_upd_txn == WT_TXN_ABORTED", session->name);
+        if (check_visibility && !__wt_txn_visible_all(session, btree->max_upd_txn, WT_TS_NONE))
```

Comment:
Should we also check timestamp for global visibility? This check now only considers transaction id but ignores timestamp.

Reply:
If we only want to consider transaction id, then checking __txn_visible_all_id is enough.

Reply:
I think we should change the comment `Don't close the btree if there are changes not yet globally visible.` to indicate that we are actually checking if there is any uncommitted update in the btree.

Reply:
I'm not sure what wording you want. The present wording matches the function __wt_txn_visible_all().
I changed it that sentence to: 

Don't close the btree if there are uncommitted updates, i.e. if there are changes not yet globally visible.

If you don't like that, please propose something.

Reply:
I would suggest "Abort if there is uncommitted updates on the btree. We check global visibility because we may not have a snapshot."

Reply:
Done. I changed it to "Abort if there are uncommitted update(s) on the btree. We check global visibility instead of snapshot visibility because we might not have a snapshot."